workflow/usage.rst

```diff
@@ -0,0 +1,228 @@
+.. index::
+    single: Workflow; Usage
+
+How to Use the Workflow
+=======================
+
+Using the Workflow component will help you to keep your domain logic as
+configuration. Having domain logic in one place gives you a better overview
+and it is easier to maintain whenever the domain requirement changes since
+you do not have to edit all your controllers, twig templates and services.
+
+A workflow is a process or a lifecycle that your objects go through. Each
+step or stage in the process is called a *place*. You do also define *transitions*
+to that describes the action to get from one place to another.
+
+.. image:: /_images/components/workflow/states_transitions.png
+
+A set of places and transitions creates a **definition**. A workflow needs
+a ``Definition`` and a way to write the states to the objects (i.e. an
+instance of a :class:`Symfony\\Component\\Workflow\\MarkingStore\\MarkingStoreInterface`.)
+
+Consider the following example for a blog post. A post can have places:
+'draft', 'review', 'rejected', 'published'. You can define the workflow
+like this:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        framework:
+            workflows:
+                blog_publishing:
+                    type: 'workflow' # or 'state_machine'
+                    marking_store:
+                        type: 'property_accessor' # or 'scalar'
+                        arguments:
+                            - 'currentPlace'
+                    supports:
+                        - AppBundle\Entity\BlogPost
+                    places:
+                        - draft
+                        - review
+                        - rejected
+                        - published
+                    transitions:
+                        to_review:
+                            from: draft
+                            to:   review
+                        publish:
+                            from: review
+                            to:   published
+                        reject:
+                            from: review
+                            to:   rejected
+
+    .. code-block:: xml
+
+        <!-- app/config/config.xml -->
+        <?xml version="1.0" encoding="utf-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd"
+        >
+
+            <framework:config>
+                <framework:workflow name="blog_publishing" type="workflow">
+                    <framework:marking-store type="scalar">
+                      <framework:arguments>currentPlace</framework:arguments>
```

Comment:
this should be `<framework:argument>currentPlace</framework:argument>` if https://github.com/symfony/symfony/pull/20458 is merged